### PR TITLE
Make manager tasks' interval and start date optional and non-defaulted

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:** 
StartDate and Interval fields are made optional and non-defaulted to make the validation more reasonable, while not changing the default behaviour.

This PR also moves manager's scheduler task spec conversion to a shared func.

**Which issue is resolved by this Pull Request:**
Prerequisite for https://github.com/scylladb/scylla-operator/pull/1851.

/kind feature
/priority important-longterm
/cc

### CODE REVIEW

**Review by zimnx — COMMENTED** [review]
lgtm, please squash last two non-generated commits, as "Move manager task validation to validating webhook" also contains changes to existing tests that "React to schema and validation changes in existing tests " should have based on the title.

**Comment by rzetelskik** [comment]
> lgtm, please squash last two non-generated commits, as "Move manager task validation to validating webhook" also contains changes to existing tests that "React to schema and validation changes in existing tests " should have based on the title.

done, thanks

**Comment by rzetelskik** [comment]
Changing the scope of this PR due to https://github.com/scylladb/scylla-operator/issues/1867#issuecomment-2033936731

**Review by tnozicka — APPROVED** [review]
one nit, rest lgtm
/approve

File: test/e2e/set/scyllacluster/scyllamanager.go [hunk 1]
```diff
@@ -136,7 +137,7 @@ var _ = g.Describe("Scylla Manager integration", func() {
 		repairTask := tasks.TaskListItemSlice[0]
 		o.Expect(repairTask.Name).To(o.Equal(sc.Status.Repairs[0].Name))
 		o.Expect(repairTask.ID).To(o.Equal(sc.Status.Repairs[0].ID))
-		o.Expect(repairTask.Schedule.Interval).To(o.Equal(sc.Status.Repairs[0].Interval))
+		o.Expect(repairTask.Schedule.Interval).To(o.Equal(*sc.Status.Repairs[0].Interval))
```
[the remark this answers is not in the document]
Comment (rzetelskik) [reply test/e2e/set/scyllacluster/scyllamanager.go:141]: done

**Review by zimnx — APPROVED** [review]
/lgtm

**Comment by rzetelskik** [comment]
> @rzetelskik: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> ci/prow/e2e-gke-parallel-clusterip	[a3fed6a](https://github.com/scylladb/scylla-operator/commit/a3fed6ab3336fb5faf8181979cdf40c9bc0b9050)	[link](https://prow.scylla-operator.scylladb.com/view/gs/scylla-operator-prow/pr-logs/pull/scylladb_scylla-operator/1868/pull-scylla-operator-master-e2e-gke-parallel-clusterip/1779871557737254912)	true	`/test e2e-gke-parallel-clusterip`
> [Full PR test history](https://prow.scylla-operator.scylladb.com/pr-history?org=scylladb&repo=scylla-operator&pr=1868). [Your PR dashboard](https://prow.scylla-operator.scylladb.com/pr?query=is:pr+state:open+author:rzetelskik).

https://prow.scylla-operator.scylladb.com/view/gs/scylla-operator-prow/pr-logs/pull/scylladb_scylla-operator/1868/pull-scylla-operator-master-e2e-gke-parallel-clusterip/1779871557737254912#1:test-build-log.txt%3A1619
```
• [FAILED] [1020.830 seconds]
Scylla Manager integration [It] should register cluster and sync repair tasks
github.com/scylladb/scylla-operator/test/e2e/set/scyllacluster/scyllamanager.go:29
...
  STEP: Waiting for repair to finish @ 04/15/24 14:25:57.016
  [FAILED] in [It] - github.com/scylladb/scylla-operator/test/e2e/set/scyllacluster/scyllamanager.go:190 @ 04/15/24 14:35:57.023
```

Controller logs:
```
2024-04-15T14:25:55.960499577Z I0415 14:25:55.960239       1 manager/sync.go:134] "Executing action" action="add task &{ClusterID: Enabled:true ID: Name:weekly Properties:map[intensity:1 parallel:123 small_table_threshold:1073741824] Schedule:0xc000483340 Tags:[] Type:repair}"
...
2024-04-15T14:25:56.175142345Z I0415 14:25:56.172692       1 manager/controller.go:148] "Hit conflict, will retry in a bit" Key="e2e-test-scyllacluster-psxjg-q2tff/basic-hl65b" Error="Operation cannot be fulfilled on scyllaclusters.scylla.scylladb.com \"basic-hl65b\": the object has been modified; please apply your changes to the latest version and try again"
...
2024-04-15T14:25:56.253308196Z I0415 14:25:56.253239       1 manager/sync.go:134] "Executing action" action="delete task \"b3dc792a-587a-43c8-818f-cad6bf869934\""
...
2024-04-15T14:25:56.361381890Z I0415 14:25:56.361067       1 manager/sync.go:134] "Executing action" action="add task &{ClusterID: Enabled:true ID: Name:weekly Properties:map[intensity:1 parallel:123 small_table_threshold:1073741824] Schedule:0xc00081e310 Tags:[] Type:repair}"
```

Known issue: https://github.com/scylladb/scylla-operator/issues/1752

https://github.com/scylladb/scylla-operator/issues/1694#issuecomment-2057253969
/retest

**Comment by rzetelskik** [comment]
> @rzetelskik: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> ci/prow/e2e-gke-parallel-clusterip	[a3fed6a](https://github.com/scylladb/scylla-operator/commit/a3fed6ab3336fb5faf8181979cdf40c9bc0b9050)	[link](https://prow.scylla-operator.scylladb.com/view/gs/scylla-operator-prow/pr-logs/pull/scylladb_scylla-operator/1868/pull-scylla-operator-master-e2e-gke-parallel-clusterip/1779908412608352256)	true	`/test e2e-gke-parallel-clusterip`
> [Full PR test history](https://prow.scylla-operator.scylladb.com/pr-history?org=scylladb&repo=scylla-operator&pr=1868). [Your PR dashboard](https://prow.scylla-operator.scylladb.com/pr?query=is:pr+state:open+author:rzetelskik).

image was already gone

/test images
/retest

**Pull request merged**